orient_express/deployment.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import argparse
+from kfp import compiler, dsl
+
+import conf
+import google.cloud.aiplatform as aip
+
+
+def delete_old_schedules(schedule_name):
+    for schedule in aip.PipelineJobSchedule.list(
+        filter=f'display_name="{schedule_name}"', order_by="create_time desc"
+    ):
+        logging.info(f"Deleting the old schedule {schedule}")
+        schedule.delete()
+
+
+def deploy_pipeline(
+    run_type,
+    pipeline_dsl,
+    pipeline_root,
+    pipeline_name,
+    pipeline_display_name,
+    pipeline_schedule_name,
+    gcp_project,
+    gcp_location,
+    gcp_service_account,
+    gcp_network,
+    gcp_labels={},
+    cron_string="*/15 * * * *",
```

Comment:
it is better to default single run as default instead of defaulting a cron string which will make run_type with default='single-run'. This will force the caller to provide an expected schedule than scheduling every job at the same time with the default time.

Reply:
run_type is required parameter, which will explicitly ask what you want - single run or deploying a scheduler